dask/highlevelgraph.py

```diff
@@ -394,7 +394,7 @@ def __dask_distributed_pack__(
 
         # Calculate dependencies without re-calculating already known dependencies
         # - Start with known dependencies
-        dependencies = known_key_dependencies.copy()
+        dependencies = copy.deepcopy(known_key_dependencies)
```

Comment:
1. If the input is actually a read-only mapping, this will return... another read-only mapping that you can't write to on line 409.
2. there is no reason to spend extra CPU for a deep copy.
```suggestion
        dependencies = ensure_dict(known_key_dependencies, copy=True)
```